app/models/ckey2bibframe.rb

```diff
@@ -0,0 +1,38 @@
+# SSH to Symphony and run a catalogdump on a ckey
+class Ckey2bibframe
+  include ActiveModel::Model
+  include ActiveModel::Validations
+  attr_accessor :ckey, :baseuri, :idfield, :marcxml
+  validates :ckey, presence: { message: ': You must enter a known ckey!' }
+
+  def catalogdump
+    Net::SSH.start(Settings.symphony_host,
+                   'sirsi',
+                   auth_methods: %w(publickey),
+                   host_key: 'ssh-rsa') do |ssh|
+      @result = ssh.exec!("source #{Settings.symphony_env} && echo #{ckey} |
+                           #{Settings.symphony_catalogdump}")
+    end
+  rescue => e
+    Rails.logger.warn('Unable to connect to Symphony via SSH.')
+    Rails.logger.error(e)
+  end
+
+  def marc21_to_xml
+    jar = 'lib/xform-marc21-to-xml-jar-with-dependencies.jar'
+    # Java flag fixes issue with JDBC Connection Reset when using many processes on 64 bit system
+    # See https://community.oracle.com/thread/943911
+    marcxml = `printf '#{catalogdump}' |
+               java -cp #{jar} edu.stanford.MarcToXMLStream -Djava.security.egd=file:/dev/../dev/urandom`
+    File.write("tmp/#{ckey}.xml", marcxml)
+    Nokogiri::XML(marcxml, &:noblanks)
+  end
+
+  def marcxml_to_bibframe
+    bf2_xsl = 'loc_marc2bibframe2/xsl/marc2bibframe2.xsl'
+    marcxml_file = "tmp/#{ckey}.xml"
+    bibframe = `xsltproc --stringparam baseuri #{baseuri} #{bf2_xsl} #{marcxml_file}`
+    FileUtils.rm(marcxml_file)
+    Nokogiri::XML(bibframe, &:noblanks)
+  end
```

Comment:
See link above about running a system command and capture of the STDOUT, STDERR etc.